src/main/java/com/minecolonies/coremod/entity/ai/citizen/guard/AbstractEntityAIGuard.java

```diff
@@ -1,654 +1,650 @@
+        {
+            currentAttackDelay--;
+            return GUARD_ATTACK_RANGED;
+        }
+        else
+        {
+            currentAttackDelay = getAttackDelay();
+        }
 
-        int tries = 0;
-        BlockPos pos = null;
-        while (pos == null
-                 || world.getBlockState(pos).getMaterial().isLiquid()
-                 || !world.getBlockState(pos.down()).getMaterial().isSolid()
-                 || (!world.isAirBlock(pos) && !world.isAirBlock(pos.up())))
+        if (getOwnBuilding() != null && worker.getCitizenData() != null)
         {
-            final Tuple<EnumFacing, EnumFacing> direction = getRandomDirectionTuple(random);
-            pos =
-              new BlockPos(worker.getPosition())
-                .offset(direction.getFirst(), random.nextInt(LENGTH_RANGE))
-                .offset(direction.getSecond(), random.nextInt(LENGTH_RANGE))
-                .up(random.nextInt(UP_DOWN_RANGE))
-                .down(random.nextInt(UP_DOWN_RANGE));
-            lastDirection = direction;
-
-            if (tries >= MAX_TRIES)
+
+            if (worker.getDistance(target) > getAttackRange())
             {
-                return this.getOwnBuilding().getLocation();
+                worker.isWorkerAtSiteWithMove(target.getPosition(), getAttackRange());
+                return GUARD_ATTACK_RANGED;
             }
 
-            tries++;
-        }
+            final int bowslot = InventoryUtils.getFirstSlotOfItemHandlerContainingTool(new InvWrapper(getInventory()),
+              ToolType.BOW,
+              0,
+              getOwnBuilding().getMaxToolLevel());
 
+            if (bowslot != -1)
+            {
+                worker.setHeldItem(EnumHand.MAIN_HAND, bowslot);
 
-        if (BlockPosUtil.getDistance2D(pos, this.getOwnBuilding().getLocation()) > ((AbstractBuildingGuards) getOwnBuilding()).getPatrolDistance())
-        {
-            return this.getOwnBuilding().getLocation();
+                worker.faceEntity(target, (float) TURN_AROUND, (float) TURN_AROUND);
+                worker.getLookHelper().setLookPositionWithEntity(target, (float) TURN_AROUND, (float) TURN_AROUND);
+
+                worker.swingArm(EnumHand.MAIN_HAND);
+
+                final EntityTippedArrow arrow = new GuardArrow(world, worker);
+                final double xVector = target.posX - worker.posX;
+                final double yVector = target.getEntityBoundingBox().minY + target.height / getAimHeight() - arrow.posY;
+                final double zVector = target.posZ - worker.posZ;
+                final double distance = (double) MathHelper.sqrt(xVector * xVector + zVector * zVector);
+                double damage = getRangedAttackDamage();
+                final double chance = HIT_CHANCE_DIVIDER / (worker.getCitizenData().getLevel() + 1);
+
+                arrow.shoot(xVector, yVector + distance * RANGED_AIM_SLIGHTLY_HIGHER_MULTIPLIER, zVector, (float) 1.6D, (float) chance);
```

Comment:
![MAJOR](https://sonarsource.github.io/sonar-github/severity-major.png 'Severity: MAJOR') Assign this magic number 1.6D to a well-named constant, and use the constant instead. [![rule](https://sonarsource.github.io/sonar-github/rule.png)](https://sonar.minecolonies.com/coding_rules#rule_key=squid%3AS109)